src/mount.rs

```diff
@@ -49,6 +49,11 @@ pub fn mount_root(options: &CmdlineOptions) -> Result<()> {
         return Err("root= not found in /proc/cmdline".into());
     }
 
+    if options.rootfstype != Some("nfs".to_string()) {
```

Comment:
If the `rootfstype` is `9p` then there is no real device as well, so we need to check for that too.